### PR TITLE
Issue #12 Create MLflow experiment if it doesn't exist yet

### LINKED ISSUE #12: Create MLflow experiment if it doesn't exist yet
Currently, the pipeline expects the experiment to be created

### PR DESCRIPTION
**Pull request by michalzelechowski-getindata**
On pipeline start, mlflow experiment is created, or the existing one is used if it is not in DELETED state.

Resolves #12

### CODE REVIEW

**Review by szczeles — APPROVED** [review]
Fantastic work!

**Review by em-pe — APPROVED** [review]

**Comment by michalzelechowski-getindata** [comment]
@em-pe @szczeles I have also extracted operator to external module and provide tests - can you have a look at this as well please?

File: kedro_airflow_k8s/operators/init_experiment_operator.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
+import logging
+
+from airflow.operators.python import PythonOperator
+from airflow.utils.decorators import apply_defaults
+
+
+class InitExperimentOperator(PythonOperator):
```
Comment (em-pe) [inline kedro_airflow_k8s/operators/init_experiment_operator.py:7]: Rename to `StartMlflowExperimentOperator` or similar. Let's make it obvious it's Mlflow related

**Review by em-pe — COMMENTED** [review]
I've updated codeclimate config so it complains less, you may want to go through the [issues](https://codeclimate.com/github/getindata/kedro-airflow-k8s/pull/21) though and add relevant docstrings. 

On one hand I'm a bit worried what would happen if we use multiple versions of the plugin in one airflow environment, on the other this code shouldn't change frequently and it's much cleaner solution then duplicating the code in between the dags, so I'm good with the changes.

File: kedro_airflow_k8s/operators/init_experiment_operator.py (hunk 1 above)
Comment (michalzelechowski-getindata) [reply kedro_airflow_k8s/operators/init_experiment_operator.py]: If classes are loaded in the same scope we can try putting them inside function of DAG definitions.

**Comment by szczeles** [comment]
I like the changes as well! :rocket:

File: kedro_airflow_k8s/operators/init_experiment_operator.py (hunk 1 above)
Comment (michalzelechowski-getindata) [reply kedro_airflow_k8s/operators/init_experiment_operator.py]: Regarding the broken tests, there was update of transitive dependencies between fields, I believe it was kedro. It triggered sqlalchemy updated which turned to be incompatible with airflow. I fixed that, but also added deptree for better visibility on such issues.

**Pull request merged**